core/templates/services/autogenerated-audio-player.service.ts

```diff
@@ -34,16 +33,19 @@ export class AutogeneratedAudioPlayerService {
 
   // Not all browsers support SpeechSynthesisUtterance, so we need the
   // check to prevent a "SpeechSynthesisUtterance is not defined" error.
-  _speechSynthesis: SpeechSynthesis =
+  _speechSynthesis: SpeechSynthesis | null =
     (window.hasOwnProperty('speechSynthesis')) ? window.speechSynthesis : null;
-  utterance: SpeechSynthesisUtterance =
+  utterance: SpeechSynthesisUtterance | null =
     (window.hasOwnProperty(
       'speechSynthesis')) ? new SpeechSynthesisUtterance() : null;
 
   _play(
       html: string, language: string,
       audioFinishedCallback: () => void): void {
     this.speechSynthesisChunkerService.cancel();
+    if (this.utterance === null) {
+      return;
+    }
     var message =
       this.speechSynthesisChunkerService.convertToSpeakableText(html);
     this.utterance.text = message;
```

Comment:
Can you instead of check for when utterance is not null?
```suggestion
    if (this.utterance !== null) {
      var message =
        this.speechSynthesisChunkerService.convertToSpeakableText(html);
      this.utterance.text = message;
       .......//code...
     }
```

Reply:
https://github.com/oppia/oppia/pull/14043#pullrequestreview-780902302
these changes are suggested by above review.

Reply:
As per offline discussion with @Rijuta-s 
We found both the patterns are fine here though in the general case we will prefer having the existing pattern because of the following good practices:
1. Avoid having a lot of branching.
2. Return as early as possible.

@Rijuta-s Thanks for raising this comment! :)

Reply:
Thanks for clearing this ambiguity.